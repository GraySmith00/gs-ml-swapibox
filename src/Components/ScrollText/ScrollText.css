.scrolltext-container {
  width: 100%;
  font-family: 'Pathway Gothic One', sans-serif;
  overflow: hidden;
  background-color: rgba(0, 0, 0, 0.9);
  color: #fff;
  font-size: 1.5rem;
  height: 4rem;
  letter-spacing: 2px; }

@keyframes scrolltext-container {
  0% {
    -webkit-transform: translate3d(100%, 0, 0);
    transform: translate3d(100%, 0, 0);
    visibility: visible; }
  100% {
    -webkit-transform: translate3d(-100%, 0, 0);
    transform: translate3d(-100%, 0, 0); } }

footer {
  display: flex;
  align-items: center;
  position: fixed;
  bottom: 0;
  width: 100%; }

.crawl {
  display: flex;
  align-items: center;
  height: 100%;
  margin: auto;
  white-space: nowrap;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-timing-function: linear;
  animation-timing-function: linear;
  -webkit-animation-name: scrolltext-container;
  animation-name: scrolltext-container;
  -webkit-animation-duration: 40s;
  animation-duration: 40s; }
