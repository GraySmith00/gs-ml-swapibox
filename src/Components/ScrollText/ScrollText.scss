.scrolltext-container {
  width: 100%;
  font-family: 'Pathway Gothic One', sans-serif;
  overflow: hidden;
  background-color: rgba(0, 0, 0, 0.5);
  color: #fff;
  font-size: 150%;
  height: 5rem;
  letter-spacing: 2px;
}

@keyframes scrolltext-container {
  0% {
    -webkit-transform: translate3d(100%, 0, 0);
    transform: translate3d(100%, 0, 0);
    visibility: visible;
  }
  100% {
    -webkit-transform: translate3d(-100%, 0, 0);
    transform: translate3d(-100%, 0, 0);
  }
}

footer {
  position: fixed;
  bottom: 0;
  width: 100%;
}

.crawl {
  display: inline-block;
  white-space: nowrap;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-timing-function: linear;
  animation-timing-function: linear;
  -webkit-animation-name: scrolltext-container;
  animation-name: scrolltext-container;
  -webkit-animation-duration: 7s;
  animation-duration: 40s;
}
