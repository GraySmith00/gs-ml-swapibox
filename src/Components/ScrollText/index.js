import React, { Component } from 'react';
import './ScrollText.css';
import { filmFetchCall } from '../../helpers';

class ScrollText extends Component {
  constructor() {
    super();
    this.state = {
      title: '',
      date: '',
      quote: ''
    };
  }

  componentDidMount() {
    this.setScrollState();
  }

  setScrollState = async () => {
    const { title, date, quote } = await filmFetchCall();
    this.setState({ title, date, quote });
  };

  render() {
    const { title, date, quote } = this.state;
    return (
      <section className="scrolltext-container">
        <div className="crawl">
          <p>{quote}</p>
          <div className="film-info">
            <p>{title}</p>
            <h1>{date}</h1>
          </div>
        </div>
      </section>
    );
  }
}

export default ScrollText;
