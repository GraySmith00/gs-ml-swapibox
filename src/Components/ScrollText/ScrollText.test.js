import React from 'react';
import ScrollText from './index';
import { mount } from 'enzyme';

describe('ScrollText Component', () => {
  let wrapper;
  let mockFilm;
  let mockFilmResponse;

  beforeEach(() => {
    wrapper = mount(<ScrollText />);
    mockFilmResponse = {
      results: [
        {
          release_date: '1980-05-17',
          title: 'The Empire Strikes Back',
          opening_crawl: 'It is a dark time for the Rebellion.'
        }
      ]
    };
    mockFilm = {
      title: 'The Empire Strikes Back',
      date: '1980-05-17',
      quote: 'It is a dark time for the Rebellion.'
    };
  });

  it('should set state of title, date, quote is the response is ok', async () => {
    window.fetch = jest.fn().mockImplementation(() =>
      Promise.resolve({
        json: () => Promise.resolve(mockFilmResponse)
      })
    );
    const expected = { ...mockFilm, errors: '' };
    await wrapper.instance().setScrollState();
    expect(wrapper.state()).toEqual(expected);
  });

  it('should set the error state if the response was not ok', async () => {
    const expected = new Error('failed to fetch');
    window.fetch = jest
      .fn()
      .mockImplementation(() => Promise.reject(new Error('failed to fetch')));
    await wrapper.instance().setScrollState();
    expect(wrapper.state().errors).toEqual(expected.message);
  });
});
