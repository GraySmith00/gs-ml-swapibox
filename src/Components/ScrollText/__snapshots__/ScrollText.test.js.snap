// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollText Component should match the snapshot 1`] = `"<section class=\\"scrolltext-container\\"><div class=\\"crawl\\"><p></p><div class=\\"film-info\\"><p></p><h1></h1></div></div></section>"`;
